nextcord/state.py

```diff
@@ -881,8 +921,21 @@ async def discover_application_commands(
             fixed_guild_id = int(temp) if (temp := raw_response.get("guild_id", None)) else None
             payload_type = raw_response["type"] if "type" in raw_response else 1
 
-            response_signature = (raw_response["name"], int(payload_type), fixed_guild_id)
-            if app_cmd := self.get_application_command_from_signature(*response_signature):
+            response_signature = {
+                "type": int(payload_type),
+                "qualified_name": raw_response["name"],
+                "guild_id": fixed_guild_id,
+            }
+            app_cmd = self.get_application_command_from_signature(**response_signature)
+            if app_cmd:
+                if not isinstance(app_cmd, BaseApplicationCommand):
```

Comment:
When would `get_application_command_from_signature` not return an instance of `BaseApplicationCommand`?
Remember that instances of a class that are a subclass of the second argument for `isinstance` still returns `True`:
```python
>>> class A: pass
... 
>>> class B(A): pass
... 
>>> b = B()
>>> isinstance(b, A)
True
>>>
```

Reply:
That was added because pyright/lance was complaining about some attr. But that was in Nextcord v2.1/2 and I haven't checked if something changed regarding to that.

Reply:
From my own testing, pyright is complaining that `SlashApplicationSubcommand` is incompatible with `BaseApplicationCommand` and `SlashApplicationSubcommand` not having certain attributes and all of that stuff. If the received command is supposed to be a `BaseApplicationCommand` and not a `SlashApplicationSubcommand`, then this is fine to keep in.

Reply:
I also looked around the code, and found out that `SlashApplicationSubcommand` is not a subclass of `BaseApplicationCommand`. I thought it was for some reason.